### ISSUE TITLE
clr-icon contains svg which title is not localized

### ISSUE THREAD

**Issue by iivanovgit** [vmware/clarity#1326]
[ x ] bug
[ ] feature request
[ ] enhancement
```

### Expected behavior
Localize title or remove title from all SVGs.

### Actual behavior
Title appears only in English.

### Reproduction of behavior
Hover on any clr-icon.

### Environment details

* **Angular version:** 4.0.X

* **Clarity version:** 

* **OS and version:** Windows10

* **Browser:** [Firefox 55.0.2]

**Comment by adityarb88** [comment]
Thanks for reporting this. We will take a look at this and get back to you soon.

**Comment by mathisscott** [comment]
This enhancement presents unique challenges. Clarity icons are not angular components so the idea of an input does not apply to them.

Presently, the title is included in the SVG `title` tag. To localize this tag, we would need to have a new attribute available to the `clr-icon` custom element. If present, the value of this attribute would be switched for the default value in the `title` tag. I'm not sure if a string replace prior to initiating the custom element HTML replacement or a DOM manipulation would be better here — although I am inclined to think that the string replacement would work best.

The unique challenge is with values passed to this attribute that do not represent  strings. That's where this gets complicated. We will need to insure that the values here are properly sanitized.

This may lead us to also reconsider how these SVG are generated and loaded per the discussions we've had re: #407.

**Comment by mathisscott** [comment]
The plan here is to add an attribute to `clr-icon` that, if present, can be used to replace the text inside the svg's `title` tag.

Additionally, we want the custom element to inject a `span` after the `svg` inside the `clr-icon` tag that duplicates the title (either the title from the attribute or the default title).

This span should have a class that hides it on screens but makes it accessible on screenreaders.

This feature will also address an accessibility challenge with the icons that exists today as well.

**Comment by dennitsa** [comment]
Hello, is there a way to add custom title to a clr-icon at the moment? I see this has been marked as fixed, but the fix is "In progress"?

**Comment by mathisscott** [comment]
Hi @dennitsa 
Where are you seeing this is "in progress"? It was merged in to 0.10.12. See https://github.com/vmware/clarity/pull/1408

For a custom title, I believe all that you need to do is set the `title` attribute on a `clr-icon` element.

Note that is you are binding this value in Angular, you'll need to use `[attr.title]` because it's an element attribute. Not an input. `clr-icon` is a custom element, not an angular component.

**Comment by dennitsa** [comment]
Thanks @mathisscott for the solution with attr.title.
By "in progress" I was looking at "@mathisscott mathisscott added this to In Progress in Clarity Icons Release Party on Oct 4, 2017", when you open the link there: "https://github.com/vmware/clarity/projects/3#card-4974344" it is still in "In prgress" state.
